src/elements/element.rectangle.js

```diff
@@ -54,12 +56,108 @@ function getBarBounds(bar) {
 	};
 }
 
+function drawLines(ctx, lines) {
+	var i = 0;
+	var ilen = lines.length;
+	var line, prevLine;
+
+	if (!ilen) {
+		return;
+	}
+
+	prevLine = lines[0];
+	ctx.beginPath();
+	ctx.lineWidth = prevLine.w;
+
+	for (; i < ilen; ++i) {
+		line = lines[i];
+		if (prevLine.w !== line.w) {
+			ctx.stroke();
+			ctx.beginPath();
+			ctx.lineWidth = line.w;
+		}
+		if (prevLine.x2 !== line.x1 || prevLine.y2 !== line.y1) {
```

Comment:
Does anyone know if a `moveTo` that doesn't actually move (we are already there) is slower that this `if`?

Reply:
Simple enough to test: https://jsperf.com/canvas-moveto-vs-if/1
So this is a little faster.